plugins/bkpr/bookkeeper.c

```diff
@@ -1975,6 +2055,14 @@ static const struct plugin_command commands[] = {
 		"bkpr-channelsapy",
 		json_channel_apy
 	},
+	{
+		"bkpr-editdescriptionbypaymentid",
+		json_edit_desc_payment_id
+	},
+	{
+		"bkpr-editdescriptionbyoutpoint",
+		json_edit_desc_utxo
+	},
```

Comment:
The schema is saying `bkpr-editdescription` and imho it would be more elegant to have a single command and multiple, nullable parameters (one for payment hash and one for the outpoint). Inferring the type of description to edit based on the first parameter (eg "contains a colon") would be nice in the CLI but cause ugly gRPC protobufs

Reply:
commands where you have multiple params where one but only-one is necessary is subpar ime. you have to make a switch already around which position you're placing the identifier in and it makes it more likely there are bugs in the underlying implementation.

keeping it type separated out makes it easier to see what's available as a consumer, makes the error checking on input much simpler, and works better for strongly typed systems.

Reply:
> The schema is saying `bkpr-editdescription`

The original plan was to have a single 'edit-description' command with an overridden first parameter `identifier`, but given the above wrt strong-type systems, decided its better to break the functionality out into smaller chunks.

We can definitely add the higher-level `edit-description` command back in that calls each of these sub-commands independently if desired later.

Reply:
Yeah, we've gone simpler since we had to write schemas for everything!